pkg/provider/core.go

```diff
@@ -75,35 +76,36 @@ func (p *Provider) CreateMachine(ctx context.Context, req *driver.CreateMachineR
 		return nil, err
 	}
 
-	m, err := p.initDriver(req.Secret)
+	m, err := p.initClient(req.Secret)
 	if err != nil {
 		klog.Error(err.Error())
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	networks := []metalgo.MachineAllocationNetwork{
-		{
-			Autoacquire: true,
-			NetworkID:   providerSpec.Network,
-		},
+	networks := []*models.V1MachineAllocationNetwork{}
+	net := models.V1MachineAllocationNetwork{
+		Autoacquire: pointer.BoolPtr(true),
+		Networkid:   &providerSpec.Network,
 	}
+	networks = append(networks, &net)
```

Comment:
You can still do as before?

```suggestion
	networks := []*models.V1MachineAllocationNetwork{
		{
			Autoacquire: pointer.BoolPtr(true),
			Networkid:   &providerSpec.Network,
		},
	}
```